hooks/useQueue.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+/**
+ * Custom hook for creating and managing Queues.
+ *
+ * @param {Array} initialValue Initial value of the queue.
+ * @return {Array} Array containing the queue, enqueue and dequeue functions respectively.
+ */
+const useQueue = (initialValue) => {
+  const [queue, setQueue] = useState(initialValue);
+
+  /**
+   * Function to add a value to the rear of the queue.
+   *
+   * @param {any} value Value to be added to the queue.
+   * @returns {undefined} This function returns nothing.
+   */
+  const enqueue = (value) => {
+    setQueue([value, ...queue]);
+  };
+
+  /**
+   * Function to remove the value in the front of the queue.
+   *
+   * @returns {any} The value removed from the queue.
+   */
+  const dequeue = () => {
```

Comment:
Also the `dequeue()` should remove the element from the front. In this case, it is poping the last element.